quinn-proto/src/tests/util.rs

```diff
@@ -153,13 +203,20 @@ impl Pair {
             if let Some(ref socket) = self.server.socket {
                 socket.send_to(&buffer, packet.destination).unwrap();
             }
-            if self.client.addr == packet.destination {
+            let server_addr = match &self.routes {
+                Some(table) => table.resolve_server_to_client(packet.destination),
+                None => (self.client.addr == packet.destination).then_some(self.server.addr),
+            };
+            if let Some(server_addr) = server_addr {
                 let ecn = set_congestion_experienced(packet.ecn, self.congestion_experienced);
                 self.client.inbound.push_back((
                     self.time + self.latency,
                     ecn,
                     buffer.as_ref().into(),
+                    server_addr,
                 ));
+            } else {
+                warn!(?packet.destination, "packet from server to client lost");
```

Comment:
should this be an error?

Reply:
No, it's totally fine for this to happen in tests. We just want to know that this happened in our tests. Should I downgrade this to a debug perhaps?